paper/VIT/train_vit.py

```diff
@@ -0,0 +1,225 @@
+parser.add_argument("--mem_samples", default=100, type=int)
+parser.add_argument("--data_root", default="data", type=str)
+args = parser.parse_args()
+torch.manual_seed(args.seed)
+np.random.seed(args.seed)
+args.benchmark = True 
+args.gpus = torch.cuda.device_count()
+args.num_workers = 4*args.gpus if args.gpus else 8
+args.is_cls_token = True if not args.off_cls_token else False
+
+
+import numpy as np
+import random
+import os 
+from typing import List
+def split_dataset(dataset:torch.utils.data.Dataset,train_size:int,val_size:int,seed:int)->List[torch.utils.data.Dataset]:
+    """ Function to split a dataset
+    Args:
+        dataset (torch.utils.data.Dataset): dataset to be splitted
+        train_size (int): Number of samples to keep in training dataset
+        val_size (int): Number of samples to keep in validation dataset
+        seed (int): Seed to fix split for reproducibility
+    Returns:
+        List[torch.utils.data.Dataset]: The subsets of dataset chosen for
+            training dataset and validation dataset 
+    """
+    first_split = len(dataset) - val_size
+    data_rest, val_dataset = torch.utils.data.random_split(dataset,[first_split,val_size],generator=torch.Generator().manual_seed(seed))
+    size_train = min(len(data_rest),train_size)
+    train_dataset, _ = torch.utils.data.random_split(data_rest,[size_train,len(data_rest)-size_train], generator=torch.Generator().manual_seed(seed))
+    return train_dataset, val_dataset
+def set_seed(seed):
+    """ Method to set seed across runs to ensure reproducibility.
+    It fixes seed for single-gpu machines.
+
+    Args:
+        seed (int): Seed to fix reproducibility. It should different for
+            each run 
+
+    Returns:
+        RandomState: fixed random state to initialize dataset iterators
+    """
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False # set to false for reproducibility, True to boost performance
+    torch.manual_seed(seed)
+    np.random.seed(seed)
+    torch.cuda.manual_seed(seed)
+    random.seed(seed)
+    random_state = random.getstate()
+    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
+    return random_state
+
+def get_loaders_args(dataset_name, dataset_dir, batch_size_train, batch_size_test, train_examples, mem_examples, seed=42):
+    if dataset_name == 'c10':
+        name = 'CIFAR10'
+    elif dataset_name == 'svhn':
+        name = 'SVHN'
+    load_dataset = getattr(datasets, 'get_'+name)
+    loaders = load_dataset(dataset_dir,batch_size_train=batch_size_train, batch_size_test=batch_size_test,batch_size_memory=mem_examples,size_train=train_examples,seed=seed)
+    return loaders
+
+if not args.gpus:
+    args.precision=32
+
+if args.mlp_hidden != args.hidden*4:
+    print(f"[INFO] In original paper, mlp_hidden(CURRENT:{args.mlp_hidden}) is set to: {args.hidden*4}(={args.hidden}*4)")
+
+train_ds, test_ds = get_dataset(args)
+
+
+class Net(pl.LightningModule):
```

Comment:
Typing?

Reply:
Nice catch! Removed in [e4422db](https://github.com/KRLGroup/memory-wrap/pull/3/commits/e4422dbc6ec5c511eb0973dbb323501a3ed9a57d)